pkg/build/graph/analysis/bc.go

```diff
@@ -98,3 +102,71 @@ func FindCircularBuilds(g osgraph.Graph) []osgraph.Marker {
 
 	return markers
 }
+
+func FindPendingTags(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+ist:
+	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
+		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+		if bcNode, points := buildPointsToTag(g, uncastIstNode); points && !istNode.Found() {
+			// A build config points to the non existent tag but the latest build has failed.
+			// Since the tag isn't yet populated, there hasn't been any successful build so far.
+			if buildNode, failed := latestBuildFailed(g, bcNode); failed {
+				markers = append(markers, osgraph.Marker{
+					Node:         buildNode,
+					RelatedNodes: []graph.Node{uncastIstNode, bcNode, buildNode},
+
+					Severity:   osgraph.ErrorSeverity,
+					Key:        LatestBuildFailedErr,
+					Message:    fmt.Sprintf("%s has failed.", buildNode.(*buildgraph.BuildNode).ResourceString()),
+					Suggestion: osgraph.Suggestion(fmt.Sprintf("Inspect the build failure with 'oc logs %s'", buildNode.(*buildgraph.BuildNode).ResourceString())),
+				})
+				continue ist
```

Comment:
Every time we hit a marker for an object, there is no need to continue inspecting it (assuming markers in the same unit are placed in order of precedence).